src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java

```diff
@@ -86,7 +86,7 @@ public void testConfigurePipelineWithInvalidBasicParam() {
       List<ValidationFailure> failures = ve.getFailures();
       //Reference name throws 2 failure, one from our method and another from CDAP Util class.
       //Other 4 properties are kept blank
-      Assert.assertEquals("Failures size does not match", 6, failures.size());
+      Assert.assertEquals("Failures size does not match", 8, failures.size());
```

Comment:
It's hard to figure out what caused these 8 validation failures. Please split up this test in smaller sub-tests that each test validation of 1 config property. Someone reading this code should be able to understand exactly what is being tested.

Reply:
done